Marlin/src/lcd/ultralcd.cpp

```diff
@@ -1438,6 +1438,22 @@ void MarlinUI::update() {
 
   #endif
 
+  #if ENABLED(TOUCH_BUTTONS)
+    void MarlinUI::screen_click(uint8_t row, uint8_t col, uint8_t x, uint8_t y) {
+      //menu window, go direct to where user touch
+      if (screen_items > 0) {
+        encoderDiff = (row - encoderPosition + encoderTopLine) * ENCODER_PULSES_PER_STEP;
+      }
+      //confirm and other windows, right screen is +, left screen is -
+      else {
+        encoderDiff = row * ENCODER_PULSES_PER_STEP;
```

Comment:
I believe this doesn't need to include `row` in the calculation. One "row" worth of pulses is enough to indicate the encoder step.

Reply:
I see that you could use the "row" as a way to increment or decrement by a larger value. I'm not sure that users will be able to see this nuance, and they might think there is a malfunction.

Reply:
When screen has itens (one per line), a use row to find the current steps needed to set the position touched by the user.

When screen has no itens (like confirm screen), I use col to know if user touched left or right.

Reply:
The two "modes" of touch are exactly in the gifs. I was thinking about one for edit screens, but didn't work yet.
And, of course, we can have on mode for direct touch on target areas, like touch on icons in the status screen. (dint yet too).